tests/network/wireguard.pm

```diff
@@ -63,19 +78,53 @@ sub run {
     barrier_wait 'VPN_ESTABLISHED';
     assert_script_run "ping -c10 $vpn_remote";
 
-    zypper_call 'in iperf';
-
-    if (get_var('IS_MM_SERVER')) {
+    if (is_server) {
         assert_script_run "iperf3 --bind $vpn_local --server --daemon --port 5001";
         script_retry 'ss -lptn | grep 5001', delay => 3, retry => 3;
         mutex_create 'server_ready';
-        wait_for_children;
     } else {
         mutex_unlock 'server_ready';
         script_retry "iperf3 --bind $vpn_local --time 30 --client $vpn_remote --port 5001", timeout => 60, delay => 3, retry => 3;
     }
+    barrier_wait('IPERF_COMPLETED');
 
     assert_script_run 'ip link set down dev wg0';
+    assert_script_run 'ip link delete dev wg0';
+
+    ## Test wg-quick
+    assert_script_run('cd /etc/wireguard');
+    if (is_server) {
+        # Prepare new keys
+        assert_script_run('wg genkey | tee server | wg pubkey > server.pub');
+        assert_script_run('wg genkey | tee client1 | wg pubkey > client1.pub');
+        assert_script_run('wg genkey | tee client2 | wg pubkey > client2.pub');
+        assert_script_run('ip a && ip r');
+        exec_and_insert_password("scp -o StrictHostKeyChecking=no server.pub client* $remote:/etc/wireguard/");
+        # Prepare configuration script
+        assert_script_run('echo -e "[Interface]\nPrivateKey = `cat /etc/wireguard/server`\nAddress = ' . "$vpn_local\n" . 'ListenPort = 51820\n" > /etc/wireguard/wg0.conf');
+        assert_script_run('echo -e "[Peer]\nPublicKey = `cat /etc/wireguard/client1.pub`\nAllowedIPs = 192.168.2.2\nPersistentKeepalive = 25\n" >> /etc/wireguard/wg0.conf');
+        assert_script_run('echo -e "[Peer]\nPublicKey = `cat /etc/wireguard/client2.pub`\nAllowedIPs = 192.168.2.3\nPersistentKeepalive = 25\n" >> /etc/wireguard/wg0.conf');
```

Comment:
There is only one client and more modifications would be needed if we add another one.

Reply:
Now there are two client configurations. I suggest two, so we can check if the server is able to handle multiple client configurations

Reply:
But did you try to actually schedule second client? :smile:

Reply:
Lines 115-125. Two clients are run sequentially - activating them in parallel won't work, as their subnets collide. I didn't tried to run two client instances at once. I only test if the server is able to handle two client configurations, not if the clients are run at the same time.

Reply:
Aha, I see! Can you then please rename the interfaces / config files so the numbers are corresponding and add some empty lies and comments like
```perl
# client1
start_wgquick("wg1");

# client2
start_wgquick("wg2");
```

Reply:
Done